fetcher/src/job/job.utils.ts

```diff
@@ -92,23 +117,55 @@ function validateAdapter(adapter): IAdapter {
   }
 }
 
+function selectProxy(url: string, proxies) {
```

Comment:
I tried to rewrite it so we don't have to use global variables. Tested in [typescript playground](https://www.typescriptlang.org/play?ssl=28&ssc=2&pln=2&pc=1#code/FAFwngDgpgBACgJwPYA8wwLwwN7BjACyQGcQAuGUhASwDsBzGAHxgFdaATKAMzqg7wwISBORi1WAWwBGUBMzacefAQF9gwbuwDGIaklqUoAGyi7EqMADFaACgjIU1KMQoW0AbQC6AShyDTEBhjAEMQFxAASQ5MGAAGAKggh0sAZSRWBG0oAFkQiApsDyJSCio6ei8KCRk5VVjsdUEtWl19QwB3BHzoBFtM4zKQGgYfN0d0XHx8bQNSSgys2CxbWigOmABVACUAGX6EYx8fADoSkEEZueSJ1OoAL2WhR2diE9MGEAJLmGpuGHstweywwWDifim03wCCSmUM2HOFHYXF4aw4ABohCIxMjlGj1FCmlC-gDiItsr9DCk0OlMtk8hAIT98NSwLSlgyPGS6VAvLFAWlybl8lyhXyANQwACMfgApM80sCfvUTMRYJCoaz2fSRdylnysKFwqRoszgmEItF+UbLTFJTKYPKtUrCRooTCQHCFU4XB4tULOXrsl4vIICTAPV6uj05MAmrNaPNqa9Yh4ESQxAAiAggEAFAD0+foSCQ9FMJ1mkkzmOEogoAA44qpMenSjBs7mC-naCEAG5yCtIKs17ENpuhhPzNWmcwTWLTswgdzWOzJlw+DSTpDl4yl2wL2eWWwdvNkQvSahknvUQdV47ALc7vcHpcTY850-ny8GEI3yuZ+9Hygd5nxMRdl3fTsz3zAgTF3W8AI3ICQPofcwMPNBIM-fMLyvX8EOOIA) and seems to be working fine.

```typescript
type Proxy = {
  host: string | undefined
  port: number | undefined
}

function selectProxyFn(proxies: Proxy[]) {
  let latestId = 0
  let proxySourceMap: {[host: string]: number} = {}

  function wrapper(url: string): Proxy {
    const source = (new URL(url)).host
    const proxySize = proxies.length
    if (proxySize === 0) {
      return {host: undefined, port: undefined}
    }

    if (source in proxySourceMap) {
      proxySourceMap[source] = (proxySourceMap[source] + 1) % proxySize
    } else {
      proxySourceMap[source] = latestId
      latestId = (latestId + 1) % proxySize
    }

    return proxies[proxySourceMap[source]]
  }
  return wrapper
}
```